src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java

```diff
@@ -264,7 +265,14 @@ private static String getEndpoint(String service, String method) {
         new ThriftRequest.Builder<>(options.getServiceName(), endpoint);
     // Create a mutable hashmap for headers, as tchannel.tracing.PrefixedHeadersCarrier assumes
     // that it can call put directly to add new stuffs (e.g. traces).
-    builder.setHeaders(new HashMap<>(thriftHeaders));
+    final HashMap<String, String> headers = new HashMap<>(thriftHeaders);
+    if (this.options.getAuthProvider() != null) {
```

Comment:
I believe once we have grpc that needs to be extended with TLS case as Temporal did: https://github.com/temporalio/sdk-go/blob/master/internal/grpc_dialer.go#L73

Reply:
cc @vytautas-karpavicius

Reply:
Yes, it should be extendable to use header in gRPC.